pages/showcase.tsx

```diff
@@ -0,0 +1,292 @@
+import { useState, useCallback, useMemo } from 'react'
+import {
+  Grid,
+  GridItem,
+  Box,
+  Flex,
+  Heading,
+  Text,
+  Link,
+  HStack,
+} from '@chakra-ui/layout'
+import type { ResponsiveValue } from '@chakra-ui/react'
+import { TabList, Tabs, Tab, TabPanels, TabPanel } from '@chakra-ui/tabs'
+import { SkipNavContent, SkipNavLink } from '@chakra-ui/skip-nav'
+import { useColorModeValue } from '@chakra-ui/color-mode'
+import { LinkIcon, Icon } from '@chakra-ui/icons'
+import SEO from 'components/seo'
+import Header from 'components/header'
+import ChakraNextImage from 'components/chakra-next-image'
+import DiscordStrip from 'components/discord-strip'
+import { AdBanner } from 'components/chakra-pro/ad-banner'
+import Footer from 'components/footer'
+import { t } from 'utils/i18n'
+import _ from 'lodash'
+import showcaseData from '../configs/showcase.json'
+import type { IShowcase, ShowcaseItem } from '../scripts/get-showcase-data'
+
+const categories = Object.keys(showcaseData as IShowcase)
+const categoriesWithAll = ['all', ...categories]
+
+const allItems: ShowcaseItem[] = categories.reduce((acc, cur) => {
+  return [...acc, ...showcaseData[cur]]
+}, [])
+
+const Showcase = () => {
+  const [index, setIndex] = useState<number>(0)
+
+  const handleTabsChange = useCallback((index: number) => {
+    if (index < 0 || index >= categories.length) setIndex(0)
+    setIndex(index)
+  }, [])
+
+  return (
+    <>
+      <SEO
+        title={t('showcase.seo.title')}
+        description={t('showcase.seo.description')}
+      />
+      <SkipNavLink zIndex={20}>Skip to Content</SkipNavLink>
+      <AdBanner />
+      <Header />
+      <Box mt={20} mb={10}>
+        <SkipNavContent />
+        <Flex mx='auto' flexDir='column' alignItems='center'>
+          <Heading fontSize={{ base: '2xl', lg: '4xl' }} lineHeight='1.2'>
+            {t('showcase.title')}
+          </Heading>
+          <Text
+            maxW='560px'
+            mx='auto'
+            color={useColorModeValue('gray.500', 'gray.400')}
+            fontSize={{ base: 'lg', lg: 'xl' }}
+            mt='3'
+          >
+            {t('showcase.message')}
+          </Text>
+        </Flex>
+        <Box mt='10' mb='20' mx='auto' maxWidth={1440}>
+          <Tabs
+            size='md'
+            variant='soft-rounded'
+            colorScheme='green'
+            align={'center'}
+            isLazy={true}
+            index={index}
+            onChange={handleTabsChange}
+          >
+            <TabList w='100%' maxW='calc(100% - 3rem)' flexWrap='wrap'>
+              {categoriesWithAll.map((c) => (
+                <Tab key={c}>{_.capitalize(c)}</Tab>
+              ))}
+            </TabList>
+            <TabPanels mt='10'>
+              {useMemo(
+                () =>
+                  categoriesWithAll.map((c) => {
+                    const items: ShowcaseItem[] =
+                      c === 'all' ? allItems : showcaseData[c]
```

Comment:
We should take this block out of the JSX world. It's not going to break in this case, but the [Rule of Hooks](https://reactjs.org/docs/hooks-rules.html) suggests that we should only call hooks at the top level. Especially a memoization hook.